src/mscorlib/src/System/String.Comparison.cs

```diff
@@ -817,153 +817,106 @@ public override bool Equals(Object obj)
-                throw new ArgumentException(Environment.GetResourceString("NotSupported_StringComparison"), nameof(comparisonType));
-            Contract.EndContractBlock();
-
-            if ((Object)a==(Object)b) {
-                return true;
+            else if (comparisonType == StringComparison.CurrentCultureIgnoreCase)
+            {
+                return (CultureInfo.CurrentCulture.CompareInfo.Compare(a, b, CompareOptions.IgnoreCase) == 0);
             }
-    
-            if ((Object)a==null || (Object)b==null) {
-                return false;
+            else if (comparisonType == StringComparison.InvariantCulture)
+            {
+                return (CultureInfo.InvariantCulture.CompareInfo.Compare(a, b, CompareOptions.None) == 0);
             }
-
-            switch (comparisonType) {
-                case StringComparison.CurrentCulture:
-                    return (CultureInfo.CurrentCulture.CompareInfo.Compare(a, b, CompareOptions.None) == 0);
-
-                case StringComparison.CurrentCultureIgnoreCase:
-                    return (CultureInfo.CurrentCulture.CompareInfo.Compare(a, b, CompareOptions.IgnoreCase) == 0);
-
-                case StringComparison.InvariantCulture:
-                    return (CultureInfo.InvariantCulture.CompareInfo.Compare(a, b, CompareOptions.None) == 0);
-
-                case StringComparison.InvariantCultureIgnoreCase:
-                    return (CultureInfo.InvariantCulture.CompareInfo.Compare(a, b, CompareOptions.IgnoreCase) == 0);
-
-                case StringComparison.Ordinal:
-                    if (a.Length != b.Length)
-                        return false;
-
-                    return EqualsHelper(a, b);
-
-                case StringComparison.OrdinalIgnoreCase:
-                    if (a.Length != b.Length)
-                        return false;
-                    else {
-                        // If both strings are ASCII strings, we can take the fast path.
-                        if (a.IsAscii() && b.IsAscii()) {
-                            return (CompareOrdinalIgnoreCaseHelper(a, b) == 0);
-                        }
-                        // Take the slow path.
-
-#if FEATURE_COREFX_GLOBALIZATION
-                        return (CompareInfo.CompareOrdinalIgnoreCase(a, 0, a.Length, b, 0, b.Length) == 0);
-#else
-                        return (TextInfo.CompareOrdinalIgnoreCase(a, b) == 0);
-#endif
-                    }
-
-                default:
-                    throw new ArgumentException(Environment.GetResourceString("NotSupported_StringComparison"), nameof(comparisonType));
+            else // comparisonType == StringComparison.InvariantCultureIgnoreCase
+            {
+                return (CultureInfo.InvariantCulture.CompareInfo.Compare(a, b, CompareOptions.IgnoreCase) == 0);
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool NotPossiblyEquals(string a, string b) {
```

Comment:
is this name a bit misleading because if called with a == b == null it will return true?
I see the calls are all proceded with `if (ReferenceEquals(a, b)) return true;` so that is already handled but perhaps it would have negligible impact if this method returned false in that case.